models/iaf_cond_alpha_mc_kinetics.cpp

```diff
@@ -0,0 +1,875 @@
+  iaf_cond_alpha_mc_kinetics& n )
+  : logger_( n )
+  , s_( 0 )
+  , c_( 0 )
+  , e_( 0 )
+{
+  // Initialization of the remaining members is deferred to
+  // init_buffers_().
+}
+
+/* ----------------------------------------------------------------
+ * Parameter and state extractions and manipulation functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::iaf_cond_alpha_mc_kinetics::Parameters_::get( DictionaryDatum& d ) const
+{
+  def< double >( d, names::V_th, V_th );
+  def< double >( d, names::V_reset, V_reset );
+  def< double >( d, names::t_ref, t_ref );
+  // changes
+  def< double >( d, names::V_max, V_max );
+  def< double >( d, names::E_Ca, E_Ca );
+  def< double >( d, names::G_Ca, G_Ca );
+  def< double >( d, names::tau_m, tau_m );
+  def< double >( d, names::tau_h, tau_h );
+  def< double >( d, names::half_m, half_m );
+  def< double >( d, names::half_h, half_h );
+  def< double >( d, names::slope_m, slope_m );
+  def< double >( d, names::slope_h, slope_h );
+  def< double >( d, names::jump_Th, jump_Th );
+  def< double >( d, names::tau_Th, tau_Th );
+  def< double >( d, names::act_flag, act_flag );
+  def< double >( d, names::reset_flag, reset_flag );
+  def< double >( d, Name( "g_sp" ), g_conn[ SOMA ] );
+  def< double >( d, Name( "g_pd" ), g_conn[ PROX ] );
+
+  // create subdictionaries for per-compartment parameters
+  for ( size_t n = 0; n < NCOMP; ++n )
+  {
+    DictionaryDatum dd = new Dictionary();
+    def< double >( dd, names::t_L, t_L[ n ] );
+    def< double >( dd, names::nt_L, nt_L[ n ] );
+    def< double >( dd, names::g_L, g_L[ n ] );
+    def< double >( dd, names::E_L, E_L[ n ] );
+    def< double >( dd, names::E_ex, E_ex[ n ] );
+    def< double >( dd, names::E_in, E_in[ n ] );
+    def< double >( dd, names::C_m, C_m[ n ] );
+    def< double >( dd, names::tau_syn_ex, tau_synE[ n ] );
+    def< double >( dd, names::tau_syn_in, tau_synI[ n ] );
+    def< double >( dd, names::I_e, I_e[ n ] );
+    // changes
+    def< double >( dd, names::tau_cur_AP, tau_curAP[ n ] );
+    def< double >( dd, names::amp_cur_AP, amp_curAP[ n ] );
+    ( *d )[ comp_names_[ n ] ] = dd;
+  }
+}
+
+void
+nest::iaf_cond_alpha_mc_kinetics::Parameters_::set( const DictionaryDatum& d )
+{
+  // allow setting the membrane potential
+  updateValue< double >( d, names::V_th, V_th );
+  updateValue< double >( d, names::V_reset, V_reset );
+  updateValue< double >( d, names::t_ref, t_ref );
+  // changes
+  updateValue< double >( d, names::V_max, V_max );
+  updateValue< double >( d, names::E_Ca, E_Ca );
+  updateValue< double >( d, names::G_Ca, G_Ca );
+  updateValue< double >( d, names::tau_m, tau_m );
+  updateValue< double >( d, names::tau_h, tau_h );
+  updateValue< double >( d, names::half_m, half_m );
+  updateValue< double >( d, names::half_h, half_h );
+  updateValue< double >( d, names::slope_m, slope_m );
+  updateValue< double >( d, names::slope_h, slope_h );
```

Comment:
slopes can be positive or negative, depending on whether it belongs to the activating or de-activating function. arithmetically 0 slope will not give errors since it belongs to the power of the exponent. it however is not desirable as a sigmoidal function will not be obtained. but this is a logical error.